department-of-veterans-affairs/caseflow
# Div/appeals 42562

Resolves [Response Letters: Inbound Ops Team Users: Can add additional response letters](https://jira.devops.va.gov/browse/APPEALS-42562)

# Description
Implemented the logic that meets the AC

## Acceptance Criteria

## Testing Plan

1. Log into Caseflow as a  Inbound Ops Team User(Any user),
2. Go to "Your Queue"
3. Click "**Correspondence Cases**" in the "Switch Views" dropdown; if it is an admin user, or else go to **Your correspondence**
4. Select any correspondence and navigate to the details page.
5. Click **Response Letter**. 
6. Verify whether the **+ Add letter** button exists or not. This button should only be visible to Inbound Ops Team users.
7. Verify **+ Add letter** is visible/selectable if there are less than 3 response letters in the correspondence.
8. Verify that **+ Add letter** is not visible/selectable if 3 response letters already exist on the correspondence
9. Click Add letter button and this should populate "Add response letter" modal.
10. Verify all the required fields and dropdown exists or not that matches the AC.
11. Verify Add Button is grayed out until all required fields are selected.
12. Once everything is selected, and the **Add** button updates and adds the Response Letter to the Correspondence Details page.


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---
<img width="1452" alt="Screenshot 2024-09-18 at 8 23 36 AM" src="https://github.com/user-attachments/assets/c96d4270-2976-4260-8338-3ba47f1495b5">
<img width="911" alt="Screenshot 2024-09-18 at 8 23 43 AM" src="https://github.com/user-attachments/assets/dc3e0f16-0027-4589-9217-1aa1bd294904">
<img width="1512" alt="Screenshot 2024-09-18 at 8 26 40 AM" src="https://github.com/user-attachments/assets/b2e21375-ec90-41fb-951f-a5602c58f2de">
<img width="1428" alt="Screenshot 2024-09-18 at 8 27 22 AM" src="https://github.com/user-attachments/assets/025f4eb7-be0d-481e-b795-c2dc058a2b5d">

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
